source/fem-dem/vans_assemblers.cc

```diff
@@ -1458,9 +1461,157 @@ template class GLSVansAssemblerSaffmanMei<3>;
+GLSVansAssemblerMagnus<dim>::calculate_particle_fluid_interactions(
   NavierStokesScratchData<dim> &scratch_data)
+{
+  // particle_number is an increment that goes from 0 to n_particles_in_cell.
+  // It is incremented at the end of the loop over particles and is used to
+  // point to the element of the vectors relative_velocity and
+  // fluid_velocity_at_particle_location corresponding to the particle being
+  // looped over.
+
+  // This implementation follows the formulation in the book "Multiphase Flows
+  // with Droplets and Particles" by Crowe et al. (2011).
+  unsigned int particle_number;
+  double       C_m = 0;
+
+  const auto &relative_velocity =
+    scratch_data.fluid_particle_relative_velocity_at_particle_location;
+  const auto &Re_p = scratch_data.Re_particle;
+
+  auto &undisturbed_flow_force = scratch_data.undisturbed_flow_force;
+
+  Tensor<1, dim> lift_force;
+
+  // Physical Properties
+  Assert(
+    scratch_data.properties_manager.density_is_constant(),
+    RequiresConstantDensity(
+      "GLSVansAssemblerMagnus<dim>::calculate_particle_fluid_interactions"));
+  const double density = scratch_data.properties_manager.density_scale;
+
+  const auto pic  = scratch_data.pic;
+  particle_number = 0;
+
+  if constexpr (dim == 2)
+    {
+      for (auto &particle : pic)
+        {
+          auto particle_properties = particle.get_properties();
+
+          // Spin parameter
+          double spin_parameter =
+            particle_properties[DEM::PropertiesIndex::dp] *
+            abs(particle_properties[DEM::PropertiesIndex::omega_z]) /
+            (2.0 * relative_velocity[particle_number].norm());
+
+          // Magnus lift coefficient
+          if (spin_parameter > 1.0 && spin_parameter < 6.0 &&
+              Re_p[particle_number] > 10.0 && Re_p[particle_number] < 140.0)
+            {
+              // Oesterlé and Dinh (1998)
+              C_m = 0.45 + (2 * spin_parameter - 0.45) *
+                             exp(-0.075 * pow(spin_parameter, 0.4) *
+                                 pow(Re_p[particle_number], 0.7));
+            }
+          else
+            C_m = 2.0 * spin_parameter;
+
+          // Magnus Lift force
+          lift_force[0] =
+            0.125 * M_PI *
+            pow(particle_properties[DEM::PropertiesIndex::dp], 2.0) * density *
+            relative_velocity[particle_number].norm() *
+            (particle_properties[DEM::PropertiesIndex::omega_z] /
+             abs(particle_properties[DEM::PropertiesIndex::omega_z]) *
+             relative_velocity[particle_number][1]);
+
+          lift_force[1] =
+            0.125 * M_PI *
+            pow(particle_properties[DEM::PropertiesIndex::dp], 2.0) * density *
+            relative_velocity[particle_number].norm() *
+            (particle_properties[DEM::PropertiesIndex::omega_z] /
+             abs(particle_properties[DEM::PropertiesIndex::omega_z]) *
+             relative_velocity[particle_number][0]);
```

Comment:
From where the 2D equations come from? I do have the Crowe, if you have the page. Is the pi/8 still applicable in 2D?